pkg/codegen/schema/loader.go

```diff
@@ -349,6 +349,15 @@ func (e *PackageReferenceVersionMismatchError) Error() string {
 	)
 }
 
+func pluginSpecFromPackageDescriptor(descriptor *PackageDescriptor) workspace.PluginSpec {
+	return workspace.PluginSpec{
+		Name:              descriptor.Name,
+		Version:           descriptor.Version,
+		PluginDownloadURL: descriptor.DownloadURL,
+		Kind:              apitype.ResourcePlugin,
+	}
+}
```

Comment:
Part of me wonders if it'd be useful to make this a `Spec()` method on the descriptor itself -- I think we have prior art for that elsewhere but could be wrong. Happy either way though.